packages/components/src/components/button/button.component.ts

```diff
@@ -6,29 +6,88 @@ import { BUTTON_COLORS, BUTTON_TYPE, BUTTON_VARIANTS, DEFAULTS } from './button.
 import { getIconSize, isValidIconSize, isValidPillSize } from './button.utils';
 
 /**
- * button component, which ...
+ * `mdc-button` is a versatile component that can be configured in various ways to suit different use cases.
  *
- * @tagname mdc-button
+ * Button Types:
+ * - **Pill button**: A button that contains text value. Commonly used for call to action, tags, or filters.
+ * - **Pill button with icons**: A button containing an icon either on the left or right side of the button.
+ * - **Icon button**: A button represented by just an icon without any text.
+ *
+ * Button Variants:
+ * - **Primary**: Solid background color.
+ * - **Secondary**: Transparent background with a solid border.
+ * - **Tertiary**: No background or border, appears as plain text but retains all button functionalities.
+ *
+ * Button Colors:
+ * - **Positive**: Green color.
+ * - **Negative**: Red color.
+ * - **Accent**: Blue color.
+ * - **Promotional**: Purple color.
+ * - **Active**: White color.
+ *
+ * Button Sizes (in REM units):
+ * - **Pill button**: 40, 32, 28, 24.
+ * - **Icon button**: 64, 52, 40, 32, 28, 24.
+ * - **Tertiary icon button**: 20.
+ *
+ * Slots:
+ * - **prefix-icon**: `<mdc-icon>` component serves as either an icon within the icon button or
+ *                   as an icon positioned to the left of the button text in a pill button with icon.
+ * - **label-text**: A component to represent the text label of the button.
+ * - **postfix-icon**: `<mdc-icon>` component serves as an icon positioned to the right
+ *                      of the button text in a pill button with icon.
  *
- * @slot default - This is a default/unnamed slot
+ * The button type is determined based on the slots provided.
  *
- * @cssprop --custom-property-name - Description of the CSS custom property
+ * @dependency mdc-icon
+ *
+ * @tagname mdc-button
+ *
+ * @slot prefix-icon - Icon positioned to the left of the button text or within an icon button.
+ * @slot default slot - Text slot of the button.
+ * @slot postfix-icon - Icon positioned to the right of the button text.
  */
 class Button extends Component {
+  /**
+   * Indicates whether the button is active.
+   */
   @property({ type: Boolean }) active: boolean = false;
 
+  /**
+   * Indicates whether the button is disabled.
+   */
   @property({ type: Boolean }) disabled: boolean = false;
 
+  /**
+   * Indicates whether the button is soft disabled.
+   * The button is currently disabled for user interaction; however, it remains focusable.
+   */
   @property({ type: Boolean, attribute: 'soft-disabled' }) softDisabled: boolean = false;
 
+  /**
+   * Aria label for the button.
+   */
   @property({ type: String, attribute: 'aria-label' }) override ariaLabel: string = '';
```

Comment:
issue: i think there is no reason to have the ariaLabel here, given it can just be passed in to a normal Lit element